File: 2744. Find Maximum Number of String Pairs/solution.py
class Solution(object):
  def maximumNumberOfStringPairs(self, words):
    """
    :type words: List[str]
    :rtype: int
    """
    mySet = set()
    count = 0

    for word in words:
      if word in mySet:
        count += 1
      else:
        mySet.add(word[::-1])

    return count

File: 2744. Find Maximum Number of String Pairs/solution.java
class Solution {
  public int maximumNumberOfStringPairs(String[] words) {
    Set<String> mySet = new HashSet<>();
    int count = 0;

    for (int index = 0; index < words.length; index++){
      String word = words[index];
      if (mySet.contains(word)){
        count += 1;
      }
      else{
        String reverseWord = "";
        for (int index2 = word.length() - 1; index2 > -1; index2--){
          reverseWord += word.charAt(index2);
        }
        mySet.add(reverseWord);
      }
    }

    return count;
  }
}